packages/backend/src/apps/scheduler/triggers/get-data-out-metadata.ts

```diff
@@ -3,12 +3,12 @@ import { IDataOutMetadata, IExecutionStep } from '@plumber/types'
 async function getDataOutMetadata(
   executionStep: IExecutionStep,
 ): Promise<IDataOutMetadata> {
-  const data = executionStep.dataOut
+  const { dataOut: data } = executionStep
   if (!data) {
```

Comment:
Note: according to type, `dataOut` is supposed to be always present so the check on `data` could be superfluous. Leaving it here as I'm not confident about the level of type safety in the codebase.

Reply:
I think it's more of a safety check cus I saw that formSG metadata does it too

Reply:
I see, thanks @m0nggh . I spoke with @ogp-weeloong about this, and the conclusion is that we think there's quite a lot of room to improve type safety in the codebase. Turning on typescript compiler's strict mode right now triggers approximately 300 build errors.

Since this seems to be a widespread issue, will table this as part of a broader discussion during sync to see what we want to do about this.

FYI @pregnantboy @stanleynguyen